app/assets/javascripts/discourse/app/widgets/hooks.js

```diff
@@ -65,6 +65,21 @@ export const WidgetMouseOverHook = buildHook(MOUSE_OVER_ATTRIBUTE_NAME);
 export const WidgetMouseOutHook = buildHook(MOUSE_OUT_ATTRIBUTE_NAME);
 export const WidgetTouchEndHook = buildHook(TOUCH_END_ATTRIBUTE_NAME);
 
+export class WidgetClickElementHook extends WidgetBaseHook {
```

Comment:
@nbianca this is a bit more complicated than the initial commit. Can you explain why we need this extra complexity? It doesn't seem necessary.